### PR TITLE
Rework 6lo enablement and reduce prj.conf files content relevantly

### PR DESCRIPTION
**Pull request by tbursztyka**

### CODE REVIEW

**Review by pfalcon — CHANGES_REQUESTED** [review]
Continuing discussion from https://github.com/zephyrproject-rtos/zephyr/pull/4211#discussion_r143221874 :

> 15.4 L2 has nothing directly related to ipv6/6lo. Comparing it to BT IPSS is actually exactly the wrong comparison: BT IPSS does specify 6lo/ipv6 for it to work. Whereas 15.4 does not.

I'm not sure exactly about fine meaning which can be in "BT IPSS does specify 6lo/ipv6 for it to work. Whereas 15.4 does not." . My patch https://github.com/zephyrproject-rtos/zephyr/pull/4211 is based on Zephyr source code, this line: https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/net/ip/l2/Kconfig#L55 . As can be seen, the option is named NET_L2_BT, and doesn't even mention IPSP/IPSS in its body. My patch applied the same line as linked above ("select NET_6LO") to the option NET_L2_IEEE802154. So, both link layers which require 6LoWPAN as of now (after my patch) express this dependency in the same consistent way.

Now I don't know which is better in this case - "select" or "default if". But if NET_L2_IEEE802154 is being changed, may I suggest that NET_L2_BT is to be changed in the same manner to preserve consistency and avoid confusion?

File: samples/net/echo_client/prj_cc2520.conf [hunk 1]
```diff
@@ -13,7 +13,6 @@ CONFIG_NET_IPV6=y
 CONFIG_NET_IPV6_RA_RDNSS=y
 CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT=2
 
-CONFIG_NET_6LO=y
 CONFIG_NET_DEBUG_6LO=n
```
Comment (nashif) [inline samples/net/echo_client/prj_cc2520.conf]: i would remove this as well, first, it is weird to have it here when the option relate to it is being removed, and then it is set to n, which should be the default anyways.

File: samples/net/echo_client/prj_cc2520.conf [hunk 2]
```diff
@@ -35,7 +34,6 @@ CONFIG_NET_STATISTICS_PERIODIC_OUTPUT=y
 
 CONFIG_NET_L2_IEEE802154=y
 CONFIG_NET_DEBUG_L2_IEEE802154=y
-CONFIG_NET_L2_IEEE802154_FRAGMENT=y
 CONFIG_NET_DEBUG_L2_IEEE802154_FRAGMENT=y
```
Comment (nashif) [inline samples/net/echo_client/prj_cc2520.conf]: ditto and for all other files below

**Review by nashif — CHANGES_REQUESTED** [review]

File: samples/net/echo_client/prj_cc2520.conf (hunk 2 above)
Comment (tbursztyka) [reply samples/net/echo_client/prj_cc2520.conf]: hum here fragment debugging is set to yes, where no is the default.

File: samples/net/echo_client/prj_cc2520.conf (hunk 1 above)
Comment (tbursztyka) [reply samples/net/echo_client/prj_cc2520.conf]: I left over yes, will remove

**Comment by tbursztyka** [comment]
@pfalcon I meant BT IPSP, will correct that. However, the "BT L2" driver is in fact an IP gateway towards the bluetooth subsystem which is the real L2 here. Afaik, you cannot do anything else but IPv6 on BT IPSP. Taking into account the name of the config option is not relevant here.

**Review by jukkar — APPROVED** [review]

**Comment by Vudentz** [comment]
I guess using the default instead of select is preferred way to select dependencies, so select should only be used externally when there is no way to use defaults?

**Comment by pfalcon** [comment]
> Afaik, you cannot do anything else but IPv6 on BT IPSP.

AFAIK, it's the same for 802.15.4. As far as I'm concerned (as a high-level network programmer), both BLE and 15.4 are just different mediums for IPv6 6LoWPAN. So, my suggestion remains - to use the same mechanism for 6LoWPAN dependency for both.

**Comment by jukkar** [comment]
> AFAIK, it's the same for 802.15.4

It is possible to have ZigBee on top of 802.15.4 which does not use 6lo. We do not and probably will not have that support in upstream thou.

I propose we use the "default y if NET_L2_IEEE802154" as done in this PR. To make things consistent, we should do the same change for BT.

**Comment by tbursztyka** [comment]
@anas @pfalcon : can you rereview this PR according to comments?

**Review by pfalcon — APPROVED** [review]
Well, I'd really like to see what @jukka suggests:

> I propose we use the "default y if NET_L2_IEEE802154" as done in this PR. To make things consistent, we should do the same change for BT.

to be a part of this PR. But can be done in another PR too.

**Comment by tbursztyka** [comment]
@Vudentz  is BT IPSP meant to possibly run without 6lo?

**Review by nashif — APPROVED** [review]

**Pull request merged**